src/cpp/session/modules/SessionSVN.cpp

```diff
@@ -733,13 +736,73 @@ Error svnRevert(const json::JsonRpcRequest& request,
    std::transform(files.begin(), files.end(), std::back_inserter(paths),
                   &resolveAliasedJsonPath);
 
-   core::system::ProcessResult result;
-   error = runSvn(ShellArgs() << "revert" << globalArgs() << "-q" <<
-                  "--depth" << "infinity" <<
-                  "--" << paths,
-                  true, &result);
+   // split into two groups -- those for which we desire
+   // a recursive revert, and those for which we desire
+   // a non-recursive revert
+   std::vector<source_control::FileWithStatus> fileStatusVector;
+   error = status(FilePath(), &fileStatusVector);
    if (error)
+   {
+      LOG_ERROR(error);
       return error;
+   }
+   
+   // build map (indexed on file path) for easy lookup
+   std::map<std::string, source_control::FileWithStatus> fileStatusMap;
+   BOOST_FOREACH(const source_control::FileWithStatus& file, fileStatusVector)
+   {
+      fileStatusMap[file.path.absolutePath()] = file;
+   }
+   
+   std::vector<FilePath> recursiveReverts;
+   std::vector<FilePath> nonRecursiveReverts;
+   BOOST_FOREACH(const FilePath& filePath, paths)
+   {
+      if (!filePath.isDirectory())
+      {
+         recursiveReverts.push_back(filePath);
+         continue;
+      }
+      
+      bool shouldRevertRecursively = false;
+      std::string key = filePath.absolutePath();
+      if (fileStatusMap.count(key))
+      {
+         const source_control::FileWithStatus& fileStatus =
+               fileStatusMap[key];
+         
+         shouldRevertRecursively = fileStatus.status.status() == "A";
```

Comment:
There is, but it appears a non-recursive revert is sufficient for reverting its deletion (alongside any files that lived within that directory).